cvat/apps/consensus/views.py

```diff
@@ -0,0 +1,474 @@
+    ConsensusConflictSerializer,
+    ConsensusReportCreateSerializer,
+    ConsensusReportSerializer,
+    ConsensusSettingsSerializer,
+)
+from cvat.apps.engine.mixins import PartialUpdateModelMixin
+from cvat.apps.engine.models import Task
+from cvat.apps.engine.serializers import RqIdSerializer
+from cvat.apps.engine.utils import get_server_url
+
+
+@extend_schema(tags=["consensus"])
+@extend_schema_view(
+    list=extend_schema(
+        summary="List annotation conflicts in a consensus report",
+        parameters=[
+            # These filters are implemented differently from others
+            OpenApiParameter(
+                "report_id",
+                type=OpenApiTypes.INT,
+                description="A simple equality filter for report id",
+            ),
+        ],
+        responses={
+            "200": ConsensusConflictSerializer(many=True),
+        },
+    ),
+)
+class ConsensusConflictsViewSet(viewsets.GenericViewSet, mixins.ListModelMixin):
+    queryset = (
+        ConsensusConflict.objects.select_related(
+            "report",
+            "report__job",
+            "report__job__segment",
+            "report__job__segment__task",
+            "report__job__segment__task__organization",
+            "report__task",
+            "report__task__organization",
+        )
+        .prefetch_related(
+            "annotation_ids",
+        )
+        .all()
+    )
+
+    iam_organization_field = [
+        "report__job__segment__task__organization",
+        "report__task__organization",
+    ]
+
+    search_fields = []
+    filter_fields = list(search_fields) + ["id", "frame", "type", "job_id", "task_id"]
+    simple_filters = set(filter_fields) - {"id"}
+    lookup_fields = {
+        "job_id": "report__job__id",
+        "task_id": "report__job__segment__task__id",  # task reports do not contain own conflicts
+    }
+    ordering_fields = list(filter_fields)
+    ordering = "-id"
+    serializer_class = ConsensusConflictSerializer
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+
+        if self.action == "list":
+            if report_id := self.request.query_params.get("report_id", None):
+                # NOTE: This filter is too complex to be implemented by other means,
+                # it has a dependency on the report type
+                try:
+                    report = ConsensusReport.objects.get(id=report_id)
+                except ConsensusReport.DoesNotExist as ex:
+                    raise NotFound(f"Report {report_id} does not exist") from ex
+
+                self.check_object_permissions(self.request, report)
+
+                if report.target == ConsensusReportTarget.TASK:
+                    queryset = queryset.filter(Q(report=report)).distinct()
+                elif report.target == ConsensusReportTarget.JOB:
+                    queryset = queryset.filter(report=report)
+                else:
+                    assert False
```

Comment:
I guess the filters should not be the same in these two cases.

Reply:
```
                if report.target == QualityReportTarget.TASK:
                    queryset = queryset.filter(
                        Q(report=report) | Q(report__parent=report)
                    ).distinct()
                elif report.target == QualityReportTarget.JOB:
                    queryset = queryset.filter(report=report)
                else:
                    assert False
```
I copied it from `QualityConflictViewSet` just remove `parent` param as we don't have it here...

Reply:
I don't see how it resolves the problem with 2 different filters doing exactly the same in different lines of code. Also, why there is no parent? Aren't job reports nested in task reports?

Reply:
Job reports have task_id as null, and the same for Task reports. No there's no nesting.